packages/auth/src/platform_browser/recaptcha/recaptcha.ts

```diff
@@ -17,9 +17,38 @@
 
 import { RecaptchaParameters } from '../../model/public_types';
 
+// reCAPTCHA v2 interface
 export interface Recaptcha {
   render: (container: HTMLElement, parameters: RecaptchaParameters) => number;
   getResponse: (id: number) => string;
   execute: (id: number) => unknown;
   reset: (id: number) => unknown;
 }
+
+export function isV2(grecaptcha: Recaptcha | GreCAPTCHATopLevel | undefined): grecaptcha is Recaptcha {
+  return grecaptcha !== undefined && (grecaptcha as Recaptcha).getResponse !== undefined;
+}
+
+// reCAPTCHA Enterprise & v3 shared interface
+ export interface GreCAPTCHATopLevel extends GreCAPTCHA {
+   enterprise: GreCAPTCHA;
+ }
+ 
+ // reCAPTCHA Enterprise interface
+ export interface GreCAPTCHA {
+   ready: (callback: () => void) => void;
```

Comment:
Would you mind following the function format on these instead?
I.e.
```ts
export interface GreCAPTCHA {
  function ready(callback: () => void): void;
  // etc
}
```

Doesn't really matter a whole lot but that's the format we use for the most part in the rest of the SDK

Reply:
Thanks for the suggestion. I prefer keep it for now as it's directly duplicated from App Check. I'm looking for a way to share these interfaces between Firebase Auth and App Check and we can make these changes there.

Reply:
I didn't realize this implementation would be shared. We should probably put the implementation in the `@firebase/util` package or a new package altogether instead. @hsubox76 what do you think?

Reply:
Depending on timelines we can put it entirely in auth for now then pull it into a separate package later for sharing

Reply:
Yeah, let's put it in auth first. @hsubox76 I think making the interface and implementation shared makes it less possible to conflict with App Check. Let me know if you have any ideas.